src/sdam/state/mod.rs

```diff
@@ -204,12 +204,31 @@ impl Topology {
         criteria: &SelectionCriteria,
     ) -> Result<Option<SelectedServer>> {
         let topology_state = self.state.read().await;
-
-        server_selection::attempt_to_select_server(
+        let selected = server_selection::attempt_to_select_server(
             criteria,
             &topology_state.description,
             &topology_state.servers,
-        )
+        )?;
+        let opts = &self.common.options;
+        if let (Some(true), Some(set_name), Some(sel_server)) =
```

Comment:
This would be _way_ easier to write and read if https://rust-lang.github.io/rfcs/2497-if-let-chains.html was implemented :(